src/ConformalTracking.cc

```diff
@@ -1667,6 +1667,371 @@ void ConformalTracking::extendHighPT(UniqueKDTracks& conformalTracks, SharedKDCl
+          if (nhit->used()) {
+            streamlog_out(DEBUG9) << "-- used" << std::endl;
+            continue;
+          }
+
+          // Check that the hit is not in the opposite side of the detector (if endcap)
+          if (nhit->endcap() && kdhit->endcap() && (nhit->forward() != kdhit->forward())) {
+            streamlog_out(DEBUG9) << "-- opposite side of detector" << std::endl;
+            continue;
+          }
+
+          // Check that radial conditions are met
+          if ((vertexToTracker && nhit->getR() >= kdhit->getR()) || (!vertexToTracker && nhit->getR() <= kdhit->getR())) {
+            streamlog_out(DEBUG9) << "-- radial conditions not met" << std::endl;
+            continue;
+          }
+
+          // Check that the hit is not wildly away from the track (make cell and check angle)
+          Cell   extensionCell(conformalTracks[currentTrack]->m_clusters[nclusters - 1], nhit);
+          double cellAngle      = seedCell->getAngle(extensionCell);
+          double cellAngleRZ    = seedCell->getAngleRZ(extensionCell);
+          double maxCellAngle   = parameters._maxCellAngle;
+          double maxCellAngleRZ = parameters._maxCellAngleRZ;
+
+          if (cellAngle > maxCellAngle || cellAngleRZ > maxCellAngleRZ) {
+            streamlog_out(DEBUG9) << "-- killed by cell angle cut" << std::endl;
+            continue;
+          }
+
+          // Now fit the track with the new hit and check the increase in chi2 - use a tempTrack object: add hit, fit, get chi2, remove hit
+          double deltaChi2(0.), deltaChi2zs(0.);
+          streamlog_out(DEBUG9) << "-- tempTrack has " << tempTrack->m_clusters.size() << " hits " << std::endl;
+          tempTrack->add(nhit);
+          tempTrack->linearRegression();
+          tempTrack->linearRegressionConformal();
+          double newchi2   = tempTrack->chi2ndof();
+          double newchi2zs = tempTrack->chi2ndofZS();
+          streamlog_out(DEBUG9) << "-- tempTrack has now " << tempTrack->m_clusters.size() << " hits " << std::endl;
+          deltaChi2   = fabs(newchi2 - chi2);
+          deltaChi2zs = fabs(newchi2zs - chi2zs);
+          streamlog_out(DEBUG9) << "-- hit was fitted and has a delta chi2 of " << deltaChi2 << " and delta chi2zs of "
+                                << deltaChi2zs << std::endl;
+          tempTrack->remove(tempTrack->m_clusters.size());
+          streamlog_out(DEBUG9) << "-- tempTrack has now " << tempTrack->m_clusters.size() << " hits " << std::endl;
+
+          double chi2cut = parameters._chi2cut;
+          if (deltaChi2 > chi2cut || deltaChi2zs > chi2cut) {
+            streamlog_out(DEBUG9) << "-- killed by chi2 cut" << std::endl;
+            continue;
+          }
+          streamlog_out(DEBUG9) << "-- valid candidate" << std::endl;
+
+          bestClustersWithChi2[nhit] = deltaChi2;
+
+          // bestCluster still empty - fill it with the first candidate
+          if (bestCluster == NULL) {
```

Comment:
Here you could merge the two ifs
if(bestCluster == NULL || deltaChi2 < bestChi2){
  bestCluster = nhit;
  bestChi2 = deltaChi2;
}
else
  continue;